www/taggame

```diff
@@ -66,6 +66,11 @@ for ($i = 0, $tags = array() ; ; $i++) {
 
 // make sure we process any persistent login state
 $userid = checkPersistentLogin();
+if (!$userid && $xml) {
+    $username = get_req_data("username");
+    $password = get_req_data("password");
+    list($userid, $errCode, $errMsg) = doLogin($db, $username, $password);
```

Comment:
I think a login failure should return immediately, before attempting to do anything else.

So move the `if (!$userid) { sendResponse("401 Unauthorized") }` here.

Is there a reason not to do that...?